### ISSUE TITLE
java.io.ByteArrayInputStream behaves differently between JS and JVM

### ISSUE THREAD

**Issue by lihaoyi** [scala-js/scala-js#3913]
The following snippet:

```scala
val arr = new Array[Byte](5)
val bais = new java.io.ByteArrayInputStream("abcde".getBytes)
bais.read(arr, 0, 5)
println(bais.read(arr, 0, 0))
```

Prints `-1` on the JDK11 and `0` on Scala.JS 0.6.31.

It seems the JVM implementation goes against the spec of `InputStream#read`:

> <p> If <code>len</code> is zero, then no bytes are read and
>      <code>0</code> is returned; otherwise, there is an attempt to read at
>      least one byte. If no byte is available because the stream is at end of
>     file, the value <code>-1</code> is returned; otherwise, at least one
>    byte is read and stored into <code>b</code>.

Not sure who we should be following - the spec or the implementation - but it would be nice if it did the same thing on both platforms (Haven't tried other versions of the JDK)

**Comment by sjrd** [comment]
If it clearly goes against the spec, it's worth checking if a bug report exists for OpenJDK, and if not, open one. I've successfully got things fixed upstream that way in the past.

**Comment by gzm0** [comment]
Hah, this might be a specification bug in `ByteArrayInputStream`. The doc of `read` says:

> Reads up to len bytes of data into an array of bytes from this input stream. If pos equals count, then -1 is returned to indicate end of file. Otherwise, the number k of bytes read is equal to the smaller of len and count-pos. If k is positive, then bytes buf[pos] through buf[pos+k-1] are copied into b[off] through b[off+k-1] in the manner performed by System.arraycopy. The value k is added into pos and k is returned.

So the behavior is in spec, but the spec seems contradictory with the base class.

**Comment by gzm0** [comment]
Fixed in 5f9af7e5815836280c36bb64ecdbd0b0ce19568f
